examples/lines_bars_and_markers/scatter_hist.py

```diff
@@ -3,18 +3,62 @@
 Scatter plot with histograms
 ============================
 
-Create a scatter plot with histograms to its sides.
+Show the marginal distributions of a scatter as histograms at the sides of
+the plot.
+
+For a nice alignment of the main axes with the marginals, two options are shown
+below.
+
+* the axes positions are defined in terms of rectangles in figure coordinates
+* the axes positions are defined via a gridspec
+
+An alternative method to produce a similar figure using the ``axes_grid1``
+toolkit is shown in the
+:doc:`/gallery/axes_grid1/scatter_hist_locatable_axes` example.
+
+Let us first define a function that takes x and y data as input, as well
+as three axes, the main axes for the scatter, and two marginal axes. It will
+then create the scatter and histograms inside the provided axes.
 """
+
 import numpy as np
 import matplotlib.pyplot as plt
 
 # Fixing random state for reproducibility
 np.random.seed(19680801)
 
-# the random data
+# some random data
 x = np.random.randn(1000)
 y = np.random.randn(1000)
 
+
+def scatter_hist(x, y, axScatter, axHistx, axHisty):
```

Comment:
I find the logical cut a bit peculiar. A function plotting data into a scatter and two hist plots into three axes is quite special; in particular since the label locations and orientations of the hist plots imply additional assumptions on the axes layout.

The function is only created to not duplicate the code of filling the axes. For the two figures, which have the same layout but are created in different ways.

This makes me think: What is the purpose of the example:
1. Demonstrate this particular case of plot.
2. Demonstrate how complex layouts of multiple axes can be created in different ways.

Originally the first. IMHO adding the second makes the example more complicated to understand. It's probably better to do this in a separate example (maybe even without data) and only refer is here ("See ... for different ways of achieving the axes layout").

Reply:
I write such functions all the time, because it allows me to have the figure layout and the actual plotting separated. E.g. one could have the axes be part of an even bigger layout without changing anything within the function.

Showcasing the two ways of achieving this plot allows users to choose which one suits their needs. Initially we only had the first case of manually positionning the axes. But then users would ask how to calculate the coordinates of the rectangles in case this whole bunch is to be positionned in a 2x2 axes layout.

Reply:
unrelated: pep8-ify the variable names? (ax_histx, etc.)